partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/DynamicTyper.kt

```diff
@@ -166,10 +166,16 @@ internal class DynamicTyper {
     private fun computeDecimal(): DecimalType? {
         val (precision, scale) = types.fold((0 to 0)) { acc, staticType ->
             val decimalType = staticType as? DecimalType ?: return null
-            val constr = decimalType.precisionScaleConstraint as DecimalType.PrecisionScaleConstraint.Constrained
-            val precision = max(constr.precision, acc.first)
-            val scale = max(constr.scale, acc.second)
-            precision to scale
+            when (val constr = decimalType.precisionScaleConstraint) {
+                is DecimalType.PrecisionScaleConstraint.Constrained -> {
+                    val precision = max(constr.precision, acc.first)
+                    val scale = max(constr.scale, acc.second)
+                    precision to scale
+                }
+                is DecimalType.PrecisionScaleConstraint.Unconstrained -> {
+                    return StaticType.DECIMAL // Default constructor is unconstrained
+                }
+            }
```

Comment:
+1 on your original statement here: https://github.com/partiql/partiql-lang-kotlin/pull/1551/files#r1722324893

Hence the `is DecimalType.PrecisionScaleConstraint.Unconstrained` is a dead branch. 

Perhaps we should do something like 

```
val constr = decimalType.precisionScaleConstraint as? DecimalType.PrecisionScaleConstraint.Constrained 
                     ?: throw InternalException(...)
```

Instead of slightly pass through if somewhere there is a bug.

Reply:
Yeah, agree that if there really is a bug, we should opt to error rather than pass-through. In the latest commit [`873b6c4` (#1554)](https://github.com/partiql/partiql-lang-kotlin/pull/1554/commits/873b6c4f4ce835ded2dcc91e091cf1895c8b0f52),
- applied your suggestion to give an error
- renamed `computeDecimal` to `computeConstrainedDecimal` for clarity